### PR TITLE
Feature tx replay

### PR DESCRIPTION
**Pull request by juniuszhou**
If the ebrelayer offline for some time, we need process all missed events. the new subcommands get the events from blocks and process these events one by one.

### CODE REVIEW

**Comment by ElliotFriedman** [comment]
We are going to go with solution #1 as defined here: https://github.com/Sifchain/sifnode/issues/328#issuecomment-754794704

We will need scripts that can be fed transactions on either chain and replay them.

We will not be using an architecture where we read from block x to block y as this could result in giving someone more than they should receive if 1 transaction in the block was correct and another was not.

**Comment by ElliotFriedman** [comment]
Hi Junius, please change the structure of this replay mechanism from replaying from block x to block y to a structure where you pass the transaction id's from a certain chain, and we relay those transactions to the other chain.

**Review by ElliotFriedman — CHANGES_REQUESTED** [review]
Hi Junius, please change the structure of this replay mechanism from replaying from block x to block y to a structure where you pass the transaction id's from a certain chain, and we relay those transactions to the other chain.

**Comment by juniuszhou** [comment]
> Hi Junius, please change the structure of this replay mechanism from replaying from block x to block y to a structure where you pass the transaction id's from a certain chain, and we relay those transactions to the other chain.
That is a simple solution for replay script. If it is fine from requirement point of view, I'd like to change my replay mechanism. current solution is complicated.

File: cmd/ebrelayer/main.go [hunk 1]
```diff
@@ -189,6 +194,32 @@ func initConfig(cmd *cobra.Command) error {
 	return viper.BindPFlag(flags.FlagChainID, cmd.PersistentFlags().Lookup(flags.FlagChainID))
 }
 
+func replayEthereumCmd() *cobra.Command {
+	//nolint:lll
+	replayEthereumCmd := &cobra.Command{
+		Use:     "replayEthereum [tendermintNode] [web3Provider] [bridgeRegistryContractAddress] [validatorMoniker] [validatorMnemonic] [fromBlock] [toBlock]",
+		Short:   "replay missed ethereum events",
+		Args:    cobra.ExactArgs(9),
```
Comment (banshee) [inline cmd/ebrelayer/main.go:202]: You have comments for 7 arguments ([tendermintNode] [web3Provider] [bridgeRegistryContractAddress] [validatorMoniker] [validatorMnemonic] [fromBlock] [toBlock]), and you're requiring 9.

**Review by banshee — CHANGES_REQUESTED** [review]

File: cmd/ebrelayer/main.go (hunk 1 above)
Comment (juniuszhou) [reply cmd/ebrelayer/main.go:202]: will add the two arguments sifFromBlock and sifEndBlock

Comment (banshee) [reply cmd/ebrelayer/main.go:202]: Are we going to add any documentation about when to use this?  Do you have to go through the logs to figure out which blocks have been missed?  Can you run it more than once with the same start/end blocks?

File: cmd/ebrelayer/main.go [hunk 2]
```diff
@@ -189,6 +194,32 @@ func initConfig(cmd *cobra.Command) error {
...
+		Example: "replayEthereum tcp://localhost:26657 ws://localhost:7545/ 0x30753E4A8aad7F8597332E813735Def5dD395028 validator mnemonic --chain-id=peggy",
+		RunE:    RunReplayEthereumCmd,
+	}
+
+	return replayEthereumCmd
+}
+
+func replayCosmosCmd() *cobra.Command {
```
Comment (banshee) [inline cmd/ebrelayer/main.go:210]: Why are there two commands?  Seems like if you have to tell replayEthereumCmd the start/end blocks for both chains, then replayEthereumCmd could do everything.

Comment (juniuszhou) [reply cmd/ebrelayer/main.go:210]: We can combine the two commands into single one. It is clearer if we have seperate commands. Then you know if the message missed in Ethereum or Sifchain processed by replay script.

File: cmd/ebrelayer/main.go (hunk 1 above)
Comment (juniuszhou) [reply cmd/ebrelayer/main.go:202]: please see replay-instruction.md in docs/relayer. You can run it again, will generate some duplicated transactions. No other side effect.

File: cmd/ebrelayer/main.go (hunk 2 above)
Comment (banshee) [reply cmd/ebrelayer/main.go:210]: No need to combine them.  It just wasn't clear to me why you needed starting and ending blocks for both chains.

**Comment by khdegraaf** [comment]
This statistic is inaccurate, and you can ignore the failure as it isn't a required check.  I have a PR to try to fix or at least improve this (very difficult to test or verify without updating config in Github), but it hasn't been approved yet by @59023g .  If you update the branch by merging it with develop, the stats may update correctly.

**Review by ElliotFriedman — DISMISSED** [review]

**Review by banshee — APPROVED** [review]

**Pull request merged**